src/libs/tagging.c

```diff
@@ -506,27 +593,113 @@ static void update_sel_on_tree(GtkTreeModel *model)
       GtkTreeIter iter = parent;
       if (find_tag_iter_tagid(model, &iter, ((dt_tag_t *)tag->data)->id))
       {
-        gtk_tree_store_set(GTK_TREE_STORE(model), &iter, DT_LIB_TAGGING_TREE_COL_SEL, ((dt_tag_t *)tag->data)->select, -1);
+        gtk_tree_store_set(GTK_TREE_STORE(model), &iter, DT_LIB_TAGGING_COL_SEL, ((dt_tag_t *)tag->data)->select, -1);
         propagate_sel_to_parents(model, &iter);
       }
     }
   }
+  dt_tag_free_result(&tags);
+}
+
+// delete a tag in the tree (tree or list)
+static void delete_tree_tag(GtkTreeModel *model, GtkTreeIter *iter, gboolean tree)
+{
+  guint tagid = 0;
+  gtk_tree_model_get(model, iter, DT_LIB_TAGGING_COL_ID, &tagid, -1);
+  if (tree)
+  {
+    if (tagid)
+    {
+      gtk_tree_store_set(GTK_TREE_STORE(model), iter, DT_LIB_TAGGING_COL_SEL, 0,
+          DT_LIB_TAGGING_COL_ID, 0, DT_LIB_TAGGING_COL_COUNT, 0, -1);
+      calculate_sel_on_tree(model, iter);
+      GtkTreeIter child, parent = *iter;
+      if (!gtk_tree_model_iter_children(model, &child, &parent))
+        gtk_tree_store_remove(GTK_TREE_STORE(model), iter);
+    }
+  }
+  else
+  {
+    gtk_list_store_remove(GTK_LIST_STORE(model), iter);
+  }
+}
+
+// delete a branch of the tag tree
+static void delete_tree_path(GtkTreeModel *model, GtkTreeIter *iter, gboolean root, gboolean tree)
+{
+  if (tree)
```

Comment:
some comments on the different paths would help here I think.

Reply:
ok. done